src/app/core/dashboard/dashboard/dashboard.component.ts

```diff
@@ -33,7 +34,58 @@ import { RouteTarget } from "../../../route-target";
   standalone: true,
 })
 export class DashboardComponent implements DashboardConfig {
-  @Input() widgets: DynamicComponentConfig[] = [];
+  @Input() set widgets(widgets: DynamicComponentConfig[]) {
+    this._widgets = widgets.filter((widget) => this.userHasAccess(widget));
+  }
+  get widgets(): DynamicComponentConfig[] {
+    return this._widgets;
+  }
+  private _widgets: DynamicComponentConfig[] = [];
+
+  constructor(private ability: EntityAbility) {}
+
+  private userHasAccess(widget: DynamicComponentConfig): boolean {
+    const entity = this.getWidgetEntity(widget);
+    if (entity) {
+      if (Array.isArray(entity)) {
+        return entity.some((e) => this.ability.can("read", e));
+      } else {
+        return this.ability.can("read", entity);
+      }
+    }
+    // No entity relation -> show widget
+    return true;
+  }
+
+  /**
+   * Detect, which entity is required for which widget.
+   *
+   * TODO in the future the widget itself should expose this
```

Comment:
That would be much nicer, of course. Any idea how to get there with the dynamic components?

What if we implement this as part of `DashboardWidgetComponent` (maybe with an input the custom widgets can set, if appropriate)?

Reply:
The only way I see for this would be by loading the components here already so we can access any static members of it.